### PR TITLE
Multi parsing with unusual encodings (fix for #1174)

### PR DESCRIPTION
**Pull request by oliver-s-lee**
Just as a note, multi-file parsing and single-file parsing now both use `errors = 'ignore'`, which can potentially hide encoding errors.

### CODE REVIEW

File: cclib/parser/logfileparser.py [hunk 1]
```diff
@@ -124,6 +97,46 @@ def seek(self, pos: int, ref: int) -> None:
             self.pos = self.size
 
 
+def opencompressedfile(filename, mode = "r", object = None, wrap = False):
```
Comment (berquist) [inline cclib/parser/logfileparser.py:100]: Let's start trying to use type annotations, even if we don't check them and they aren't necessarily correct the first time, they can be refined later; I'm not sure about `object` or the return type:
```suggestion
def opencompressedfile(filename: str, mode: str = "r", object: typing.Optional[typing.IO] = None, wrap: bool = False) -> typing.IO:
```

**Review by berquist — CHANGES_REQUESTED** [review]

File: cclib/parser/logfileparser.py [hunk 2]
```diff
@@ -134,39 +147,20 @@ def openlogfile(filename: str, object=None):
...
 
         # This is needed, because fileinput will assume stdin when filename is empty.
         if len(filename) == 0:
             return None
-
-        return FileWrapper(fileinput.input(filename, openhook=fileinput.hook_compressed))
+        
+        # The 'errors' argument of fileinput.FileInput() looks like it should do what we want here,
+        # but it's only available in python3.10 and even then doesn't work properly in conjunction with openhook...
+        return fileinput.FileInput(filename, openhook= lambda filename, mode, encoding = None, errors = None: opencompressedfile(filename, mode))
```
[the remark this answers is not in the document]
Comment (oliver-s-lee) [reply cclib/parser/logfileparser.py:163]: Yeah no worries can do. It's essentially just juggling the call signature so it fits with what FileInput expects, but I think I can write it better. 

I think logging should be possible yeah, let me take a look.

File: cclib/parser/logfileparser.py (hunk 1 above)
Comment (berquist) [reply cclib/parser/logfileparser.py]: Oh, it was not so much about the argument being called `object` despite overwriting a built-in...it's just that I don't know what the type is here.

I personally annotate all functions, no matter their visibility.

**Review by berquist — APPROVED** [review]
Thanks for writing tests!

**Comment by berquist** [comment]
I'm going to make another issue to update the logging, because this is scary looking:
Screenshot from 2023-04-27 22-22-07

**Pull request merged**